### ISSUE TITLE
[MJAVADOC-591] javadoc fails with maven.compiler.release=8 and Automatic-Module-Name

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-javadoc-plugin#889]
**[Robert Varga](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=nite)** opened **[MJAVADOC-591](https://issues.apache.org/jira/browse/MJAVADOC-591?redirect=false)** and commented

In a module, which defines an Automatic-Module-Name and (inherits) maven.compiler.release=8, but does not set source=8, javadoc generation fails with:

```java
16052 [ERROR] Exit code: 2 - error: option --module-path not allowed with target 8
16052 [ERROR] error: option --add-modules not allowed with target 8
16052 [ERROR] error: option --patch-module not allowed with target 8
16053 [ERROR]
16053 [ERROR] Command line was: /home/nite/jdk-12/bin/javadoc @options @packages

```

This occurs with both  javadoc-11.0.2 and javadoc-12. Using either maven-javadoc-plugin-3.0.1, defining source=8 or removing Automatic-Module-Name works just fine.

I think the problem is that maven.compiler.release is propagated to release property, but that is not taken into account when --add-modules is generated and propagated.

---

**Affects:** 3.1.0

**Remote Links:**
- [GitHub Pull Request #25
](https://github.com/apache/maven-javadoc-plugin/pull/25)

**Comment by jira-importer** [comment]
**[Robert Varga](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=nite)** commented

Since maven.compiler.release overrides maven.compiler.source, and '–release' overrides '-source' in javac, it seems logical to default 'source' to the same value as 'release'. It certainly fixes the issue in local testing.

**Comment by jira-importer** [comment]
**[Robert Varga](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=nite)** commented

It seems to work with 3.1.1-SNAPSHOT.

**Comment by jira-importer** [comment]
**[Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte)** commented

I don't recognize any commits since 3.1.0 related to this. Shall I close this as 'can't reproduce'?

**Comment by jira-importer** [comment]
**[Robert Varga](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=nite)** commented

Ah, right, it was just my local repo that got a candidate fix and there's no 3.1.1-SNAPSHOT in http://repository.apache.org/snapshots/ and I did not notice 'mvn -U' did not clear out my local version.

A candidate fix is here: https://github.com/apache/maven-javadoc-plugin/pull/25

**Comment by jira-importer** [comment]
**[Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte)** commented

This shouldn't be required. There's a new property since 3.1.0 called [release](https://maven.apache.org/plugins/maven-javadoc-plugin/javadoc-mojo.html#release) (I just noticed the wrong `@since` value is used...) which already defaults to `maven.compiler.release`.

**Comment by jira-importer** [comment]
**[Robert Varga](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=nite)** commented

Yes, I noticed that property. I agree, that was just the first cut before I opened the project in an IDE :) The pull request is updated with a more correct solution.

**Comment by jira-importer** [comment]
**[Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte)** commented

Yes, that change makes sense.

**Comment by jira-importer** [comment]
**[Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte)** commented

Fixed in [56979750c1752c811da28ff058aebc83b1fd018b](https://gitbox.apache.org/repos/asf?p=maven-javadoc-plugin.git;a=commit;h=56979750c1752c811da28ff058aebc83b1fd018b)
Thanks for the patch!
